File: src/domain/Review.ts
import User from "./User";

export default class Review {
  constructor(
    public user: User,
    public rating: number,
    public title: string,
    public created_at: Date,
    public updated_at: Date,
    public verified_purchase: boolean,
    public helpful_votes: number,
    public id?: number,
  ) {}
}

File: src/main/java/com/sandro/wanted_shop/domain/Review.java
package com.sandro.wanted_shop.domain;

import com.sandro.wanted_shop.domain.common.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "reviews")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Review extends BaseTimeEntity {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    private Product product;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @Column(nullable = false)
    private Integer rating;

    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    private Boolean verifiedPurchase = false;

    private Integer helpfulVotes = 0;
} 